controller/api/public/stat_summary.go

```diff
@@ -356,18 +358,19 @@ func processRequests(results []promResult, groupBy model.LabelNames) map[string]
 				basicStats[label] = &pb.BasicStats{}
 			}
 
-			value := uint64(0)
-			if !math.IsNaN(float64(sample.Value)) {
-				value = uint64(math.Round(float64(sample.Value)))
-			}
+			value := extractSampleValue(sample)
 
 			switch result.prom {
 			case promRequests:
 				switch string(sample.Metric[model.LabelName("classification")]) {
 				case "success":
-					basicStats[label].SuccessCount = value
+					basicStats[label].SuccessCount += value
```

Comment:
curious why we're summing here with `+=` ? does it have something to do with discrete `secured == false` and `secured == true` metrics?

based on `metricToKey`, i'd assume the `label` value to be something like `foo/success/true` (pod/classification/secured).

Reply:
The key generated here has to match the key generated in MetaNamespaceKeyFunc, which returns resource or namespace/resource. (In resourceQuery, we look up the objects, and return a map of objects, keyed by MetaNamespaceKeyFunc).

The groupBy passed in to metricToKey doesn't include classification or meshed, because that groupBy is the one generated in buildRequestLabels (so it only has namespace and resource).

So the reason for the `+=` is that if there are discrete secured=false, or secured=true metrics, that I'd just want to collapse them all into the total request count. This also collapses secured success and failures into one total secured count, as the percentage is all we (currently) care about. This does lose some information, as you noted in your comment.

Reply:
Ah! I didn't realize the `groupBy` here only had namespace/resource.